acme/authority.go

```diff
@@ -64,22 +81,22 @@ var (
 )
 
 // NewAuthority returns a new Authority that implements the ACME interface.
-func NewAuthority(db nosql.DB, dns, prefix string, signAuth SignAuthority) (*Authority, error) {
-	if _, ok := db.(*database.SimpleDB); !ok {
+func NewAuthority(ops AuthorityOptions, signAuth SignAuthority) (*Authority, error) {
```

Comment:
`NewAuthority` is used by Caddy, I'll suggest to create a `New()` and just call `New()` in new authority.

And perhaps that new could have the arguments reversed, `New(a SignAuthority, opts AutorityOptions)`, looks more natural to have the options at the end.